### PR TITLE
fix: Drop verb client interfaces

### LINKED ISSUE #2757: Incorrect Java code generated for verbs with primitive response type
When a Java module depends on a verb with primitive response type, and non-primitive param type, the generated Java verb client doesn't compile. 

The generated client should use boxed types instead of primitives to match the `xyz.block.ftl.VerbClient` interface.

This verb:
```
//ftl:verb export
func TheVerb(ctx context.Context, val string) (int, error) {
    return 1, nil
}
```

Creates this generated client:
```
@VerbClientDefinition(
    name = "theVerb",
    module = "example"
)
public interface TheVerbClient extends VerbClient<String, Long> {
  long call(@NotNull String value);
}
```

Compilation of this client fails with: 
> [ERROR] \<module\>/TheVerbClient.java:[13,8] call(@org.jetbrains.annotations.NotNull String) in ftl.example.TheVerbClient clashes with call(P) in xyz.block.ftl.VerbClient
> [ERROR]   return type long is not compatible with java.lang.Long

### PR DESCRIPTION
**Pull request by tomdaffurn**
Drop the `VerbClient`, `VerbClientSink`, `VerbClientSource`, `VerbClientEmpty` interfaces. They conflict with verbs that use primitive params and responses.

Fixes #2757

### CODE REVIEW

File: jvm-runtime/testdata/go/gomodule/server.go [hunk 1]
```diff
@@ -280,3 +280,13 @@ func TypeEnumVerb(ctx context.Context, val AnimalWrapper) (AnimalWrapper, error)
 //func MixedEnumVerb(ctx context.Context, val Mixed) (Mixed, error) {
 //	return val, nil
 //}
+
+//ftl:verb export
+func PrimitiveResponseVerb(ctx context.Context, val string) (int, error) {
```
[the remark this answers is not in the document]
Comment (stuartwdouglas) [reply jvm-runtime/testdata/go/gomodule/server.go:285]: So looking into this I don't know if we need this change:

Currently we generate: 

```
@VerbClientDefinition(
    name = "intVerb",
    module = "gomodule"
)
public interface IntVerbClient extends VerbClient<Long, Long> {
  long call(long value);
}
```

Which is valid, even though the primitive method does not override the interface type, because we actually generate both methods:

https://github.com/TBD54566975/ftl/blob/b0465e3c2d14cb72400b8724058db1445c663619/jvm-runtime/ftl-runtime/common/deployment/src/main/java/xyz/block/ftl/deployment/VerbProcessor.java#L93

Basically in the int -> int case we generate three different methods:

Object -> Object
Long -> Long
long -> long

[the remark this answers is not in the document]
Comment (tomdaffurn) [reply jvm-runtime/testdata/go/gomodule/server.go:285]: The compile error doesn't show up for `IntVerb` because the generated signature of `call(long)` is different to `call(Long)` in `VerbClient`. Need to have an Object param with primitive response value to make the signature the same but different response type.

I guess we can restrict boxing to cases where the param is an Object and the response isn't.

[the remark this answers is not in the document]
Comment (stuartwdouglas) [reply jvm-runtime/testdata/go/gomodule/server.go:285]: Ah, I see what you mean. 

I wonder how much implementing `VerbClient` actually matters. For the generated classes I don't think it actually adds anything, and not being able to use primitives is just kinda annoying. I am not 100% happy with the way verb clients currently work anyway, as invocation within the same module is clunky. Let me think about this a bit.

[the remark this answers is not in the document]
Comment (stuartwdouglas) [reply jvm-runtime/testdata/go/gomodule/server.go:285]: Possibly something like: https://github.com/TBD54566975/ftl/compare/stuartwdouglas/jvm-api-refactor?expand=1

**Review by stuartwdouglas — APPROVED** [review]
LGTM, although I think we might want to move to method annotations at some point.

File: jvm-runtime/ftl-runtime/common/deployment/src/main/java/xyz/block/ftl/deployment/VerbProcessor.java [hunk 2]
```diff
@@ -213,21 +157,36 @@ VerbClientBuildItem handleVerbClients(CombinedIndexBuildItem index, BuildProduce
...
-                        "@VerbClientDefinition can only be applied to interfaces that directly extend a verb client type with concrete type parameters and "
-                                + iface.name() + " does not extend a verb client type");
+                clients.put(iface.name(), new VerbClientBuildItem.DiscoveredClients(name, module, cc.getClassName()));
             }
         }
         return new VerbClientBuildItem(clients);
     }
 
+    private MethodInfo getCallMethod(ClassInfo verbClient) {
+        for (var call : verbClient.methods()) {
```
Comment (stuartwdouglas) [inline jvm-runtime/ftl-runtime/common/deployment/src/main/java/xyz/block/ftl/deployment/VerbProcessor.java:169]: I think we need to throw an exception if the interface has additional non-static non-default methods, as they won't be implemented by the generated client.

Comment (tomdaffurn) [reply jvm-runtime/ftl-runtime/common/deployment/src/main/java/xyz/block/ftl/deployment/VerbProcessor.java:169]: That would only happen if the user has mistakenly created their own @VerbClient interface right?
